libafl/src/monitors/disk_aggregate.rs

```diff
@@ -0,0 +1,131 @@
+//! Monitors that log aggregated stats to disk.
+
+use alloc::vec::Vec;
+use core::{
+    fmt::{Debug, Formatter},
+    time::Duration,
+};
+use std::{fs::OpenOptions, io::Write, path::PathBuf};
+
+use libafl_bolts::{current_time, ClientId};
+use serde_json::json;
+
+use crate::monitors::{Aggregator, ClientStats, Monitor};
+
+/// A monitor that wraps another monitor and logs aggregated stats to a JSON file.
+#[derive(Clone)]
+pub struct OnDiskJsonAggregateMonitor<M>
+where
```

Comment:
you don't need `M: Monitor` in this struct definition

Reply:
also in `impl<M> Debug for OnDiskJsonAggregateMonitor<M>` and in `impl<M> OnDiskJsonAggregateMonitor<M> ` too